radixdlt-java/src/main/java/com/radixdlt/client/atommodel/quarks/ChronoQuark.java

```diff
@@ -4,37 +4,44 @@
 import com.radixdlt.client.core.atoms.particles.Quark;
 import org.radix.serialization2.DsonOutput;
 import org.radix.serialization2.SerializerId2;
+import org.radix.time.Timestamps;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 
 /**
  * A quark that keeps a specific timestamp.
  * Currently used to keep atom-specific timestamps in TimestampParticle.
  */
 @SerializerId2("CHRONOQUARK")
 public final class ChronoQuark extends Quark {
-	@JsonProperty("timestampKey")
+	@JsonProperty("timestamps")
 	@DsonOutput(DsonOutput.Output.ALL)
-	private String timestampKey;
-
-	@JsonProperty("timestampValue")
-	@DsonOutput(DsonOutput.Output.ALL)
-	private long timestampValue;
+	private Map<String, Long> timestamps = new HashMap<>();
 
 	private ChronoQuark() {
 	}
 
-	public ChronoQuark(String timestampkey, long timestampValue) {
-		this.timestampKey = Objects.requireNonNull(timestampkey);
-		this.timestampValue = timestampValue;
+	public ChronoQuark(String timestampKey, long timestampValue) {
+		this.timestamps.put(Objects.requireNonNull(timestampKey, "timestampKey is required"), timestampValue);
+	}
+
+	public long getTimestamp() {
```

Comment:
Would it make sense to have a corresponding `setTimetamp(long)` and constructor to mirror this operation?